electronics_lib/BatteryProtector_S8200A.py

```diff
@@ -0,0 +1,96 @@
+from electronics_abstract_parts import *
+
+
+class BatteryProtector_S8200A_Device(DiscreteChip, FootprintBlock):
+  def __init__(self):
+    super().__init__()
+
+    self.vss = self.Port(Ground())
+    self.vdd = self.Port(VoltageSink(
+      voltage_limits=self.vss.voltage_limits + RangeExpr._to_expr_type((-0.3, 12) * Volt),
+      current_draw=(0.7, 5.5) * uAmp
+    ))
+
+    self.vm = self.Port(Passive())
+    self.do = self.Port(Passive())
+    self.co = self.Port(Passive())
+
+
+  def contents(self):
+    super().contents()
+    self.footprint(
+      'U', 'Package_TO_SOT_SMD:SOT-23-6',
+      {
+        '1': self.do,
+        '2': self.vm,
+        '3': self.co,
+        # '4': no connection pin,
+        '5': self.vdd,
+        '6': self.vss,
+      },
+      mfr='ABLIC', part='S-8200ABE-M6T1U',
+      datasheet='https://www.mouser.com/datasheet/2/360/S8200A_E-1365901.pdf'
+    )
+
+class BatteryProtector_S8200A(Block):
+  @init_in_parent
+  def __init__(self) -> None:
+    super().__init__()
+
+    self.ic = self.Block(BatteryProtector_S8200A_Device())
+
+    self.pwr_out = self.Port(VoltageSource())
+    self.gnd_out = self.Port(GroundSource())
+    self.pwr_in = self.Port(VoltageSink(
+      voltage_limits=self.ic.vdd.voltage_limits,
```

Comment:
This actually is discarded, because the pin is connected (so it derives its properties from the inner connections.

If the system was a bit smarter about defaults processing, it would warn you if a non-default was specified.